### ISSUE TITLE
Bug: [no-unsafe-assignment] doesn't report on an `any` value assigned as an initializer of an `accessor` property

### ISSUE THREAD

**Issue by ronami** [typescript-eslint/typescript-eslint#10784]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.ts&code=MYGwhgzhAEBiD29oG8BQ1oHpPQE4FMAHeXAF3wBNpgSDhSQBPdaMaAXmgEZWYwA7RgG5ULbNH7xSeIiXJUARgFdpEABbwlIRfhZhgwfFBLQFHbr1aCRAX1RA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZolp9kySgHMmAW0Qt0URNGgdokcGAC%2BITUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class Foo {
  // reported correctly
  a = 1 as any;

  // not reported but should be
  accessor b = 1 as any;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unsafe-assignment": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected the rule to report on the `accessor` property, similar to how it reports on a standard property.

### Actual Result

The rule didn't report on the `accessor` property, even though it was assigned an `any` type.

### Additional Info

See https://github.com/typescript-eslint/typescript-eslint/issues/10763#issuecomment-2634900340.
